uber/site_sections/registration.py

```diff
@@ -251,9 +251,12 @@ def delete(self, session, id, return_to='index?'):
                 session.delete_from_group(attendee, attendee.group)
                 message = 'Unassigned badge removed.'
             else:
-                session.add(Attendee(**{attr: getattr(attendee, attr) for attr in [
+                replacement_attendee = Attendee(**{attr: getattr(attendee, attr) for attr in [
                     'group', 'registered', 'badge_type', 'badge_num', 'paid', 'amount_paid', 'amount_extra'
-                ]}))
+                ]})
+                if replacement_attendee.group and replacement_attendee.group.is_dealer:
+                    replacement_attendee.ribbon == c.DEALER_RIBBON
```

Comment:
I originally thought to just port over the ribbon along with everything else, but this would be undesirable in most cases -- you don't really want your new 'unassigned' attendee to have a Volunteer or Department Head ribbon just because the person you unassigned had one. To be frank, I'm not sure porting all that other information over is a particularly good idea either, but I didn't want to mess with it.